src/components/StaffPayment/StaffPayment.vue

```diff
@@ -172,30 +170,31 @@ export default class StaffPayment extends Vue {
 
   /** Prompt the field validations. */
   @Watch('validate')
-  private validateFields (): void {
-    this.$refs.fasForm.validate()
-    this.$refs.bcolForm.validate()
+  private validateFields (val: boolean): void {
+    if (val) {
+      if (this.paymentOption === StaffPaymentOptions.FAS) this.$refs.fasForm.validate()
+      if (this.paymentOption === StaffPaymentOptions.BCOL) this.$refs.bcolForm.validate()
+    }
   }
 
   /** Called when payment option (radio group item) has changed. */
   @Watch('paymentOption')
   private onPaymentOptionChanged (val: number): void {
-    this.validateFields()
     switch (val) {
       case StaffPaymentOptions.FAS:
         // reset other form and update data
-        this.$refs.bcolForm.reset()
+        this.$refs.bcolForm.resetValidation()
```

Comment:
When reset is called, the `@input` event fires, which causes a bunch of events/validation that we don't care about.

Conversely, when resetValidation is called, the validation errors/state are cleared and that's it. Nice and clean.